From: "Eberhard S. Amann" <esa@linux.vnet.ibm.com>
Date: Wed, 10 Sep 2014 16:37:52 +0200
Subject: [PATCH] GenWQE: Fix problem when reading HSI and Retc
Patch-mainline: v3.18-rc1
Git-commit: 08e4906cc29d092ae2da0ff089efe1488e584d3c
References: bsc#949731,LTC#131502

This patch fixes a problem we found during debug on PPC64 when
reading HSI status and Retc.

Signed-off-by: Eberhard S. Amann <esa@linux.vnet.ibm.com>
Signed-off-by: Frank Haverkamp <haver@linux.vnet.ibm.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/misc/genwqe/card_ddcb.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/misc/genwqe/card_ddcb.c
+++ b/drivers/misc/genwqe/card_ddcb.c
@@ -390,8 +390,9 @@ static int genwqe_check_ddcb_queue(struc
 		    0x00000000)
 			goto go_home; /* not completed, continue waiting */
 
-		/* Note: DDCB could be purged */
+		wmb();  /*  Add sync to decouple prev. read operations */
 
+		/* Note: DDCB could be purged */
 		req = queue->ddcb_req[queue->ddcb_act];
 		if (req == NULL) {
 			/* this occurs if DDCB is purged, not an error */
